client/src/pages/admin/mentor-registry.tsx

```diff
@@ -102,149 +134,113 @@ function Page(props: Props) {
+    activeTab === MentorRegistryTabsMode.New
+      ? setActiveTab(MentorRegistryTabsMode.All)
+      : setActiveTab(MentorRegistryTabsMode.New);
+  }, [activeTab]);
+
+  const handleModalDataChange = (mode: ModalDataMode, record: MentorRegistryDto) => {
+    setIsModalOpen(true);
+    setModalData({ mode, record });
+  };
+
+  const onCancelModal = () => {
+    setModalData(null);
+    setIsModalOpen(false);
+  };
 
-function renderInfo(_: any, record: MentorRegistryDto) {
-  const isMentor = record.courses.some(id => !record.preselectedCourses.includes(id));
   return (
-    <div className="info-icons">
-      {record.englishMentoring ? <div title="Ready to mentor in English" className="icon-flag-uk" /> : null}
-      {isMentor ? <div title="Mentor in the past" className="icon-mentor" /> : null}
-      {record.hasCertificate ? (
-        <SafetyCertificateTwoTone
-          title="Completed with certificate"
-          className="icon-certificate"
-          twoToneColor="#52c41a"
+    <AdminPageLayout
+      session={props.session}
+      title="Mentor Registry"
+      loading={loading}
+      courses={courses}
+      styles={{ margin: 0, padding: 0 }}
+    >
+      <Row justify="space-between" style={{ padding: '0 24px', minHeight: 64 }} align="bottom" className="tabs">
+        <Tabs tabBarStyle={{ margin: '0' }} activeKey={activeTab} items={tabs} onChange={handleTabChange} />
+        <Button
+          icon={<FileExcelOutlined />}
+          style={{ alignSelf: 'center' }}
+          onClick={() => (window.location.href = `/api/registry/mentors/csv`)}
+        >
+          Export CSV
+        </Button>
+        <style jsx>{styles}</style>
+      </Row>
+      <Col style={{ background: '#f0f2f5', padding: 24 }}>
+        <Alert
+          message={
+            <>
+              The number of mentors below can mentor: <Typography.Text strong>{maxStudents} students</Typography.Text>
+            </>
+          }
+          type="info"
+          showIcon
+          style={{ marginBottom: 24 }}
         />
-      ) : null}
-      <style jsx>{styles}</style>
-    </div>
+        <MentorRegistryTableContainer
+          mentors={data}
+          courses={courses}
+          activeTab={activeTab}
+          disciplines={disciplines}
+          handleModalDataChange={handleModalDataChange}
+        >
+          {mentorRegistryProps => <MentorRegistryTable {...mentorRegistryProps} />}
+        </MentorRegistryTableContainer>
+      </Col>
+      {isModalOpen && modalData?.mode === ModalDataMode.Invite && renderModal()}
+      {isModalOpen && modalData?.mode === ModalDataMode.Resend && (
+        <MentorRegistryResendModal
+          modalData={modalData || {}}
+          modalLoading={modalLoading}
+          resendConfirmation={resendConfirmation}
+          onCancel={onCancelModal}
+        />
+      )}
+      {isModalOpen && modalData?.mode === ModalDataMode.Delete && (
+        <MentorRegistryDeleteModal
+          modalData={modalData || {}}
```

Comment:
I guess there is no need to have this fallback, as modaldata in this case is always defined, right? (here and above as well)